src/org/rascalmpl/library/lang/json/internal/JsonValueReader.java

```diff
@@ -492,8 +574,16 @@ public IValue visitNode(Type type) throws IOException {
         }
         
         in.endObject();
+        int endPos = getPos();
+        int endLine = getLine();
+        int endCol = getCol();
 
-        return vf.node("object", new IValue[] { }, kws);
+        if (src != null) {
+          return vf.node("object", new IValue[] { }, kws).asWithKeywordParameters().setParameter("src", vf.sourceLocation(src, startPos, endPos - startPos + 1, startLine, endLine, startCol, endCol + 1));
```

Comment:
here the parameter is named `src`? not `location`?

Reply:
Yes because JSON is not HTML, so there is no a priori name clash there. However, it would be better if we test for a clash dynamically.